src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import axios from "axios";
+import Header from "./components/Header";
+import RepoDetailList from "./components/RepoDetails/RepoDetailList";
+import styled from "styled-components";
+
+const Container = styled.div`
+  padding: 2rem;
+`;
+
+const UserName = styled.div`
+  font-size: 2rem;
+  margin-bottom: 1rem;
+`;
+const RepoInfo = styled.div`
+  font-size: 1.5rem;
+  margin-bottom: 0.5rem;
+`;
+
+const App = () => {
+  const gitURL = "https://api.github.com/users";
+
+  const [userName, setUserName] = useState("");
+  const [repos, setRepos] = useState([]);
+  const [userInfo, setUserInfo] = useState({
+    name: userName,
+    repo: 0,
+    star: 0
+  });
+
+  const getUserName = e => {
+    e.preventDefault();
+    setUserName(e.target.value);
+  };
```

Comment:
```suggestion
  const getUserName = useCallback(e => {
    e.preventDefault();
    setUserName(e.target.value);
  }, []);
```
getUserName에 함수를 할당할 때 `useCallback` 이나 lodash의 'memoize`을 사용해 메모이제이션을 해주는게 좋을거 같아요! 메모이제이션을 해주지 않으면 username을 입력할 때마다 `App`컴포넌트는 리렌더링 되는데 그때마다 `getUserName`에도 새로운 함수가 할당되고 이를 속성으로 넘겨준 모든컴포넌트가 불필요하게 리렌더링 될 수 있을거같습니다

Reply:
처음에 `useCallback` 함수 활용해서 구현하려고 했는데 그땐 두번째 인자에 대한 이해가 부족해서 그냥 넘겼었네요ㅠㅜ 그래도 이렇게 짚어주신김에 더 찾아보니까 조금은 알게된 것 같아요😊 효율성을 위해 적절하게 사용하도록 하겠습니다!